### PR TITLE
Implement function to auto create hyperlink for ConvertTo-Html

### LINKED ISSUE #11458: Feature Request - ConvertTo-Html should auto create hyperlinks
There are many times where I've had to go manually craft an `<a href="....` as a new property where the existing string property already had a valid URL in it, just so ConvertTo-Html would generate a hyperlink. 

Its non-intuitive for something that creates HTML to be unable to recognize url's and emit hyperlinks, It doesnt have to be a default action, but it would save the time and trouble of hand building `<a href` strings

### PR DESCRIPTION
**Pull request by JJJcfff**
# PR Summary
Closes #11458.

Add a helper function that checks the property value(string) of objects passed into ConvertTo-Html. If the string is a valid absolute URI, then convert it to a hyperlink format in HTML

Add a switch "-hyperlink" to trigger the auto conversion feature.

## PR Context
This Pull Request saves the user time when creating HTML documents by auto casting URI to hyperlinks

## PR Checklist

- [ ] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [ ] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [ ] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

**Comment by JJJcfff** [comment]
@microsoft-github-policy-service agree



________________________________

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/ConvertTo-Html.cs [hunk 1]
```diff
@@ -198,6 +198,27 @@ public SwitchParameter Fragment
...
+                return _hyperlink;
+            }
+
+            set
+            {
+                _hyperlink = value;
+            }
+        }
+        
+        private SwitchParameter _hyperlink;
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/ConvertTo-Html.cs:220]: Please use auto property.

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/ConvertTo-Html.cs [hunk 2]
```diff
@@ -647,6 +668,18 @@ private void WritePropertyValue(StringBuilder Listtag, MshParameter p)
...
+        /// <summary>
+        /// Check if a string is uri, if true, then change to hyperlink format.
+        /// </summary>
+        private static string CheckUri(string s, bool hyper)
+        {
+            if (hyper && Uri.TryCreate(s, UriKind.Absolute, out Uri uriResult) && (uriResult.Scheme == Uri.UriSchemeHttp || uriResult.Scheme == Uri.UriSchemeHttps))
+            {
+                return "<a href=\"" + s + "\">" + s + "</a>";
+            }
+            return s;
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/ConvertTo-Html.cs:680]:
```suggestion
            }

            return s;
```

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/ConvertTo-Html.cs (hunk 1 above)
Comment (JJJcfff) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/ConvertTo-Html.cs]: I changed it to bool since this is ment to be a switch

**Pull request closed without merging**